content/articles/a-look-at-c#-coding-conventions/index.md

```diff
@@ -0,0 +1,238 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /a-look-at-c#-coding-conventions/
+title: A look at C# coding conventions
+description: In this tutorial, we will look at the C# language coding conventions that a programmer needs to consider and apply when programming.
+author: amos-njoroge
+date: 2021-11-18T00:00:00-12:00
+topics: [Languages]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/a-look-at-c#-coding-conventions/hero.jpg
+    alt: A look at C# coding conventions Hero Image
+---
+This article outlines a set of coding guidelines, design principles, and naming conventions for the C# programming language and the `.NET` framework. It lays out the rules in a format that is simple to read and understand, allowing software engineers to apply them rapidly.
+<!--more-->
+The article's aim is to serve as a primary source of information and efficiency for programmers.
+
+Benefits of coding conventions:
+1. They make copying, changing, and maintaining code a lot easier.
+2. They help readers understand the code faster by allowing them to make assumptions based on prior knowledge.
+3. They give the code a unified design so readers may concentrate on the content rather than the layout.
+4. They tell the best way to use the C# programming language in the most ideal manner.
+
+Several coding conventions are used in the C# programming language:
+1. Naming conventions
+2. Layout conventions
+3. Commenting conventions
+
+There are a few naming conventions that should be followed while writing C# code. All information about public elements applies as well to protected and protected internal elements in the following scenarios, which are all supposed to be visible to external callers.
+
+### Pascal's case
+In the Pascal case, every first letter of every word is capitalized as displayed in the example below. One can use the Pascal case for identifiers with three or more characters.
+
+When naming a class, record, or struct, use pascal casing ("PascalCasing").
+
+```
+PublicSchool1
+```
+
+```c#
+public class PascalCase1
+{
+}
+public record PublicIntitution1(
+    string Country,
+    string County,
+    string Village,
+    );
+public struct ValueCoordinate1
+{
+}
+```
+
+Besides to prefixing the name with an I, when naming an interface, use the pascal casing. Customers can tell it's a user interface because of this.
```

Comment:
Explain further here

Reply:
What I mean is that beside using pascal casing when naming an interface, one should also prefix the interface name with an I so that it can be easily recognized as an interface